lib/migration.c

```diff
@@ -71,7 +71,11 @@ init_migration(const vfu_migration_callbacks_t *callbacks,
     migr->pgsize = sysconf(_SC_PAGESIZE);
 
     /* FIXME this should be done in vfu_ctx_realize */
-    migr->state = VFIO_DEVICE_STATE_RUNNING;
+    if (flags & LIBVFIO_USER_MIG_FLAG_START_RESUMING) {
```

Comment:
Why do we need this? Can't the client transition from RUNNING to RESUMING?

Reply:
The client is not allowed to transition from RUNNING to RESUMING in the kernel's VFIO protocol so we don't allow it here.

Reply:
OK, then we should document the purpose of this flag where it gets defined.